packages/components/src/components/textarea/textarea.ts

```diff
@@ -247,93 +249,150 @@ export default class SdTextarea extends SolidElement implements SolidFormControl
-    const hasHelpTextSlot = this.hasSlotController.test('help-text');
-    const hasLabel = this.label ? true : !!hasLabelSlot;
-    const hasHelpText = this.helpText ? true : !!hasHelpTextSlot;
+    // Slots
+    const slots = {
+      label: this.hasSlotController.test('label'),
+      helpText: this.hasSlotController.test('help-text')
+    };
+
+    // States
+    const hasLabel = this.label ? true : !!slots['label'];
+    const hasHelpText = this.helpText ? true : !!slots['helpText'];
+    const hasValidationAttr = this.required || !!this.minlength || !!this.maxlength;
+    const isInvalid = hasValidationAttr && !this.checkValidity();
+    const isValid = hasValidationAttr && this.checkValidity();
+    // Hierarchy of input states:
+    const inputState = this.disabled
+      ? 'disabled'
+      : this.readonly
+        ? 'readonly'
+        : this.hasFocus && isInvalid
+          ? 'activeInvalid'
+          : this.hasFocus && isValid
+            ? 'activeValid'
+            : this.hasFocus
+              ? 'active'
+              : isInvalid
+                ? 'invalid'
+                : isValid
+                  ? 'valid'
+                  : 'default';
+
+    // Conditional Styles
+    const textSize = this.size === 'sm' ? 'text-sm' : 'text-base';
+    const textColor = {
+      disabled: 'text-neutral-500',
+      readonly: 'text-black',
+      activeInvalid: 'text-error',
+      activeValid: 'text-success',
+      active: 'text-black',
+      invalid: 'text-error',
+      valid: 'text-success',
+      default: 'text-black'
+    }[inputState];
+    const borderColor = {
+      disabled: 'border-neutral-500',
+      readonly: 'border-neutral-800',
+      activeInvalid: 'border-error border-2',
+      activeValid: 'border-success border-2',
+      active: 'border-primary border-2',
+      invalid: 'border-error',
+      valid: 'border-success',
+      default: 'border-neutral-800'
+    }[inputState];
 
     return html`
       <div
         part="form-control"
-        class=${classMap({
-      'form-control': true,
-      'form-control--small': this.size === 'small',
-      'form-control--medium': this.size === 'medium',
-      'form-control--large': this.size === 'large',
-      'form-control--has-label': hasLabel,
-      'form-control--has-help-text': hasHelpText
-    })}
+        class=${cx(
+          'form-control text-left',
+          hasLabel && 'form-control--has-label',
+          hasHelpText && 'form-control--has-help-text',
+          this.disabled && 'pointer-events-none cursor-not-allowed'
+        )}
       >
         <label
           part="form-control-label"
-          class="form-control__label"
+          class=${cx('form-control-label mb-2', hasLabel ? 'inline-block' : 'hidden', textSize)}
           for="input"
           aria-hidden=${hasLabel ? 'false' : 'true'}
         >
-          <slot name="label">${this.label}</slot>
+          <slot name="label">${this.label}${this.required ? '*' : ''}</slot>
```

Comment:
We should not handle different logics here. Please double-check whether we incorporate the 'required (*)' in the CSS or HTML and make the change consistently. Currently, it is displayed in the CSS for sd-radio, sd-switch, and sd-checkbox and in HTML for sd-input and sd-textarea.

Reply:
Good hint :) I applied the changes to the input as well